File: Capitulo_01/01_Getting_Started:_The_Easy_Problems/02_Easy/11559_:::_Event_Planning_★/Main.java
import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main
{
	public static void main(String[] args) throws IOException
	{
		BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
		//-----------------------------------------------------------
		int numParticip, budget, numHoteles, minGasto, precio, gasto;
		String[] paramEvento, camasDisp;

		for(String linea; (linea=br.readLine())!=null;)
		{
			paramEvento=linea.split(" ");
			numParticip=Integer.parseInt(paramEvento[0]);
			budget=Integer.parseInt(paramEvento[1]);
			numHoteles=Integer.parseInt(paramEvento[2]);
			minGasto=-1;

			while(numHoteles--!=0)
			{
				precio=Integer.parseInt(br.readLine());
				camasDisp=br.readLine().split(" ");

				gasto=precio*numParticip;
				for(String camas : camasDisp)
				{
					int beds=Integer.parseInt(camas);
					if(gasto<=budget && beds>=numParticip && (gasto<minGasto||minGasto==-1)) minGasto=gasto;
				}
			}

			String rta=(minGasto==-1)?"stay home":minGasto+"";
			System.out.println(rta);
		}		
		//-----------------------------------------------------------
		br.close();
	}
}

File: Capitulo_01/01_Getting_Started:_The_Easy_Problems/02_Easy/11559_:::_Event_Planning_★/Main.c
#include "stdio.h"

int main()
{
	int N, B, H, W, minGasto, p, gasto, beds, i, j;
	while(scanf("%d %d %d %d", &N, &B, &H, &W)==4)
	{
		minGasto=-1;
		for(i=0; i<H; i++)
		{
			scanf("%d", &p);
			gasto=N*p;

			for(j=0;j<W;j++)
			{
				scanf("%d", &beds);
				if(gasto<=B && beds>=N && (gasto<minGasto || minGasto==-1)) minGasto=gasto;
			}
		}
		if(minGasto==-1) puts("stay home");
		else printf("%d\n",minGasto);
	}

	return 0;
}
